### DISCUSSION TITLE
shared vcpkg-configuration.json for multiple projects

### DISCUSSION THREAD

**Discussion by niyue** [microsoft/vcpkg#21493] [Q&A]
Hi there,

I wonder how this should be done when using vcpkg manifest mode and a custom vcpkg registry.
1) I have a multi-module project (e.g. `parent-project`), and I separate the entire project into multiple modules, and each module is distributed as a vcpkg package (e.g. `sub-project-1`, `sub-project-2`), with its own vcpkg manifest. 
2) I create a custom vcpkg registry, installing all sub project modules into the registry. For all these modules, they share some common third party dependencies, e.g. `fmt`. 
3) Some shared dependencies are overridden in the custom vcpkg registry, for example, I want to use `fmt` version `7.0.0` instead of `8.0.0` in the latest vcpkg port definition. To use the overridden vcpkg packages, after reading the vcpkg documentation, to my knowledge, I have to create a `vcpkg-configuration.json` file along side with vcpkg manifest file `vcpkg.json`, where the custom registry and the overridden dependency is specified. 
4) I want to make all of my sub projects follow the same way to manage dependencies:
4.1) I would like all of them to use the same custom registry
4.2) I would like all of them to use the same version of overridden dependency (e.g. all of them should use `fmt` `7.0.0`) 

Instead of creating multiple `vcpkg-configuration.json` files for each sub project, is there any approach I can ask all the sub projects to share the `vcpkg-configuration.json` so that:
a) I don't have to specify the custom registry multiple times 
b) I don't have to specify the overridden packages multiple times (since they should be the same), and I can easily upgrade the shared overridden dependency to a newer version (e.g. upgrading `fmt 7.0.0` to `fmt 7.1.0`) by updating single place in a file

**Comment by mathisloge** [comment]
As I had a very similar need I think it would be good to have some merging functionalities. 
Two cases: 
1. specify the path to the "global" vcpkg-configuration.json in the vcpkg.json
2. add a project local vcpkg-configuration.json and the thing from (1). Then vcpkg should merge the local with the global to create a single vcpkg-configuration.json. So it would be possible to override some vars of the global one.
